core/execution/engine.go

```diff
@@ -1330,9 +1330,20 @@ func (e *Engine) OnTick(ctx context.Context, t time.Time) {
 		}
 		e.removeMarket(id)
 	}
+
+	// sort the marketCPStates by ID since the order we clear insurance pools
+	// changes the division when we split it across remaining markets and
+	// who the remainder ends up with if it doesn't divide equally.
+	allIDs := []string{}
+	for id := range e.marketCPStates {
+		allIDs = append(allIDs, id)
+	}
+	sort.Strings(allIDs)
+
 	// find state that should expire
-	for id, cpm := range e.marketCPStates {
+	for _, id := range allIDs {
```

Comment:
couldn't you just iterate the allMarketsCpy for the id and skip those not in the marketCPStates?

Reply:
I don't think so, `marketCPStates` contans closed/settlement markets waiting to be succeeeded so they won't be in `allMarketsCpy`.